### ISSUE TITLE
dynamodb data source global_secondary_index list missing entries

### ISSUE THREAD

**Issue by jeffcasavant** [terraform-providers/terraform-provider-aws#10171]
### Community Note

* Please vote on this issue by adding a 👍 [reaction](https://blog.github.com/2016-03-10-add-reactions-to-pull-requests-issues-and-comments/) to the original issue to help the community and maintainers prioritize this request
* Please do not leave "+1" or "me too" comments, they generate extra noise for issue followers and do not help prioritize the request
* If you are interested in working on this issue or have submitted a pull request, please leave a comment

### Terraform Version

Terraform v0.12.8

### Affected Resource(s)

* aws_dynamodb_table

### Terraform Configuration Files

```hcl
data "aws_dynamodb_table" "orders" {
  name = var.orders_dynamodb_table
}
```

### Expected Behavior

Terraform state should include both of the GSIs that are on the referenced table.

```
terraform state show module.cloud_portal.data.aws_dynamodb_table.orders
data "aws_dynamodb_table" "orders" {
    ...
    global_secondary_index = [
        {
            hash_key           = "OrganizationId"
            name               = "OrganizationIndex"
            non_key_attributes = []
            projection_type    = "ALL"
            range_key          = ""
            read_capacity      = 20
            write_capacity     = 10
        },
        {
            hash_key           = "ProductId"
            name               = "ProductIndex"
            non_key_attributes = []
            projection_type    = "ALL"
            range_key          = ""
            read_capacity      = 20
            write_capacity     = 10
        },
    ]
   ...
}
```

### Actual Behavior

Terraform state only includes one of the GSIs that are on the referenced table.

```
terraform state show module.cloud_portal.data.aws_dynamodb_table.orders
data "aws_dynamodb_table" "orders" {
    ...
    global_secondary_index = [
        {
            hash_key           = "OrganizationId"
            name               = "OrganizationIndex"
            non_key_attributes = []
            projection_type    = "ALL"
            range_key          = ""
            read_capacity      = 20
            write_capacity     = 10
        },
    ]
   ...
}
```

### Steps to Reproduce

1.  Create a dynamodb table with two GSIs
2. Reference that table in a `data "aws_dynamodb_table"` block
3. `terraform state show data.aws_dynamodb_table.<my_test_table>`
4. Find the GSI list and see that it has only one index listed.

**Comment by barrytam20** [comment]
i've submitted PR #10339 for this

**Comment by bflad** [comment]
Hi folks 👋  This issue was likely resolved in Terraform AWS Provider version 3.2.0 -- if you are still having trouble after that release, please open a new issue and we will take a fresh look. 👍
